docs/core/howto/ssl.rst

```diff
@@ -204,10 +199,40 @@ For example,
     options = CertificateOptions(..., dhParameters=dhParams)
 
 Another part of the TLS protocol which ``CertificateOptions`` can control is the version of the TLS or SSL protocol used.
-This is often called the context's "method".
+There are two ways to configure it, either by configuring a list of protocols, or by the legacy way of setting the internal context's "method".
+By default, Twisted will configure it to use TLSv1.0 or later and disable the insecure SSLv3 protocol, but this manual control over protocols can be helpful if you need to support legacy SSLv3 systems, or you wish to restrict it down to just the strongest of the TLS versions.
+
+You can give ``CertificateOptions`` a range of TLS versions you wish it to negotiate with the ``tlsProtocols`` argument in the initializer:
+
+.. code-block:: python
+
+    from twisted.internet.ssl import (CertificateOptions, TLSVersion,
+                                      TLSVersionRange)
+    options = CertificateOptions(
+        ...,
+        tlsProtocols=TLSVersionRange(TLSVersion.TLSv1_0,
+                                     TLSVersion.TLSv1_2))
+
```

Comment:
> Saying "TLSv1.0 to newest TLS, but not TLSv1.1" might be an extreme edge case but the point of the new API is to support all the edge cases.

Is it? If it is, then you're right, this API doesn't support that. But I wasn't aware that that was the design goal.

Now that I am aware, let me say that I think that's a *bad* design goal. I don't see any reason to allow the user to say "I want only SSLv2 and TLSv1.1, please". I am completely unable to come up with a scenario where that request is necessary, but "I want everything between SSLv2 and TLSv1.1 inclusive" is not a suitable replacement.

If it were *really*, *really* necessary, is there any reason that simply constructing two separate endpoints using the same protocol factory wouldn't suffice? I am entirely at a loss for the motivation for supporting all possible combinations of TLS versions.

If the only acceptable outcome for this feature is "allow all combinations of TLS versions" then I'm actually -1 on the feature until someone convinces me that it's necessary.